internal/source/image.go

```diff
@@ -190,3 +181,65 @@ func (i *Image) handleUnexpectedPod(ctx context.Context, pod *corev1.Pod) error
 	_ = i.Client.Delete(ctx, pod)
 	return fmt.Errorf("unexpected pod phase: %v", pod.Status.Phase)
 }
+
+func pendingImagePodResult(pod *corev1.Pod) *Result {
+	var messages []string
+	for _, cStatus := range append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...) {
+		if waiting := cStatus.State.Waiting; waiting != nil {
+			if waiting.Reason == "ErrImagePull" || waiting.Reason == "ImagePullBackOff" {
+				messages = append(messages, waiting.Message)
+			}
+		}
+	}
+	return &Result{State: StatePending, Message: strings.Join(messages, "; ")}
+}
+
+// addSecurityContext is responsible for taking a container and defining the
+// relevant security context values. By having a function do this, we can keep
+// that configuration easily consistent and maintainable.
+func addSecurityContext(pod *corev1.Pod) {
+	// Check that pod is defined before proceeding
+	if pod == nil {
+		return
+	}
+
+	// Add security context for overall pod
+	pod.Spec.SecurityContext = &corev1.PodSecurityContext{
+		// TODO (tyslaton): Address unpacker pod allowing root users for image sources
+		//
+		// In our current implementation, we are creating a pod that uses the image
+		// provided by an image source. This pod is not always guaranteed to run as a
+		// non-root user and thus will fail to initialize if running as root in a PSA
+		// restricted namespace due to violations. As it currently stands, our compliance
+		// with PSA is baseline which allows for pods to run as root users. However,
+		// all RukPak processes and resources, except this unpacker pod for image sources,
+		// are runnable in a PSA restricted environment. We should consider ways to make
+		// this PSA definition either configurable or workable in a restricted namespace.
```

Comment:
"As it currently stands, our compliance with PSA is baseline which allows for pods to run as root users" 

So I'm a little confused about this part.

"we are creating a pod that uses the image provided by an image source": what's an example of this kind of an image? 

My confusing is surrounding the fact that we're labeling the namespace as `enforce:baseline`, and setting `runAsNonRoot: false` (which means we'll allow containers to be run as root), but then setting `allowPrivilegeEscallation:false`. Did we use an example image that fits the criteria to test out this configuration works out for those images? 

Also, since we're deploying rukpak downstream in the platform-operator namespace, I'm guessing this means we want that namespace to be labeled as `baseline` instead of restricted [here](https://github.com/openshift/platform-operators/pull/14)?

Reply:
> As a result, I turned that feature off but left the other configuration you defined in the downstream PR outside of that. Do you have a recommendation on what we'd change here?

My guess is that `allowPrivilegeEscalation` has to be set to `true` here, and I was hoping there's an image that needs to be run as root that we can use to test out if `allowPrivilegeEscalation:false` is enough for those images.

Reply:
So this image [here](https://github.com/operator-framework/rukpak/tree/main/internal/provisioner/plain#:~:text=Create%20the%20combo%20BundleDeployment%20referencing%20the%20desired%20combo%20Bundle%20configuration%3A) runs as root, was failing with `runAsNonRoot` set to true but succeeded when I turned that to false and kept `allowPrivilegeEscalation` at false.

Reply:
Okay that answers my question then 

/lgtm 🚀

Reply:
Docs about `allowPrivilegeEscalation` are here: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

> allowPrivilegeEscalation: Controls whether a process can gain more privileges than its parent process. This bool directly controls whether the [no_new_privs](https://www.kernel.org/doc/Documentation/prctl/no_new_privs.txt) flag gets set on the container process. allowPrivilegeEscalation is always true when the container:
>  -  is run as privileged, or
>  -  has CAP_SYS_ADMIN